### PR TITLE
fix examples

### PR DESCRIPTION
**Pull request by kepta**
- This fixes the examples not working.
- Uses https://github.com/tribou/jest-serializer-path for replacing absolute paths in snapshot
- Moves the `getConfig` code to `config` folder for a more holistic black box testing

@davidtheclark for review.

### CODE REVIEW

File: testSetup.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+const serializer = require('jest-serializer-path');
+
+expect.addSnapshotSerializer(serializer);
```
Comment (davidtheclark) [inline testSetup.js]: Do you need to register this in both the config and a test setup file? I thought it was one or the other.

File: lib/config/createUrc.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    return dynamicRequireSync({
+      path: path.join(this.outputDirectory, WEBPACK_ASSETS_BASENAME),
+      deleteCache: true
+    });
+  }
+
+  getHtmlSourceFn() {
+    return dynamicRequireSync({
+      path: this.htmlSource,
+      backupPath: path.join(__dirname, '..', './default-html.js'),
```
Comment (davidtheclark) [inline lib/config/createUrc.js]: Could `htmlSource` just have a default? Would that avoid the need for this `backupPath` option?

File: bin/underreact.js [hunk 3]
```diff
@@ -2,17 +2,12 @@
...
-const webpack = require('webpack');
 
 const start = require('../commands/start');
 const build = require('../commands/build');
 const serveStatic = require('../commands/serve-static');
 const logger = require('../lib/logger');
-const { DEFAULT_CONFIG_NAME } = require('../lib/constants');
-const createConfig = require('../lib/config');
-const getDefaultConfig = require('../lib/default-underreact.config');
+const config = require('../lib/config');
```
Comment (davidtheclark) [inline bin/underreact.js]: Is this supposed to point to `createUrc`?

**Comment by davidtheclark** [comment]
There are some solid arguments for using kebab-case in filenames instead of capitalization. I believe the reason it's a long-standing convention is because capitalization in filenames *can* cause problems since operating systems handle it differently. Better to avoid getting bitten by this in the future.

File: bin/underreact.js (hunk 3 above)
Comment (kepta) [reply bin/underreact.js]: No, to fully test the entire thing (https://github.com/mapbox/underreact/pull/14/files#diff-76c595bf9e5f6dac835e81cd694d425eR8) I extracted the code in `bin/underreact` to `/config`.

File: testSetup.js (hunk 1 above)
Comment (kepta) [reply testSetup.js]: I think I am only registering it once in the test setup file.

File: lib/config/createUrc.js (hunk 2 above)
Comment (kepta) [reply lib/config/createUrc.js]: Ya that makes sense, I have made the default htmlSource in `default-underreact.config.js` point in the latest [commit](https://github.com/mapbox/underreact/pull/13/commits/d4e9968ca0635909081a894266883a9a93095a51#diff-9641d65535017283a8bbb631f8f5a48cR33)

**Pull request merged**